File: app/src/main/java/com/example/client/dialog/PrimaryDialog.java
package com.example.client.dialog;

import android.app.AlertDialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.client.R;
import com.example.client.utils.ICallBack;

public class PrimaryDialog implements View.OnClickListener{
    private TextView description;
    private TextView btnOK, btnCancel;
    private AlertDialog dialog;
    private ICallBack oKListener;
    private ICallBack cancelListener;

    public PrimaryDialog(){

    }

    public void getInstance(Context context){
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        View dialogView = LayoutInflater.from(context).inflate(R.layout.dialog_primary,null);
        builder.setView(dialogView);
        description = dialogView.findViewById(R.id.description);
        btnOK = dialogView.findViewById(R.id.btnOK);
        btnCancel = dialogView.findViewById(R.id.btnCancel);
        dialog = builder.create();
        builder.setCancelable(false);
        btnOK.setOnClickListener(this);
        btnCancel.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btnOK:
                oKListener.onCallback();
                dialog.dismiss();
                break;
            case R.id.btnCancel:
                cancelListener.onCallback();
                dialog.dismiss();
                break;

        }
    }

    public void setOKListener(ICallBack oKListener){
        this.oKListener = oKListener;
    }

    public void setCancelListener(ICallBack cancelListener){
        this.cancelListener = cancelListener;
    }

    public void setDescription(String des){
        description.setText(des);
    }

    public void show(){
        dialog.show();
    }

    public void hideBtnCancel(){
        btnCancel.setVisibility(View.GONE);
    }
}

File: app/src/main/java/com/example/client/dialog/PrimaryDialog.kt
package com.example.client.dialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.FragmentManager
import com.example.client.R
import com.example.client.base.BaseDialog

class PrimaryDialog(private val onOk:() -> Unit, private val onCancel: () -> Unit) : BaseDialog(), View.OnClickListener {
    private var title: String? = null
    private var description: String? = null
    private var tvTitle: TextView? = null
    private var tvDescription: TextView? = null
    private var tvOk: TextView? = null
    private var tvCancel: TextView? = null
    private var showCancel = true
    private var showTitle = false

    override fun initView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.dialog_primary, null)
        tvTitle = rootView?.findViewById(R.id.tv_title)
        tvDescription = rootView?.findViewById(R.id.tv_description)
        tvOk = rootView?.findViewById(R.id.tv_ok)
        tvCancel = rootView?.findViewById(R.id.tv_cancel)
        return rootView
    }

    override fun bindEvent() {
        tvOk?.setOnClickListener(this)
        tvCancel?.setOnClickListener(this)
    }

    override fun bindComponent() {
        tvTitle?.let {
            it.text = title
            if (showTitle) it.visibility = View.VISIBLE
            else it.visibility = View.GONE
        }
        tvDescription?.text = description
        tvCancel?.let {
            if (showCancel) it.visibility = View.VISIBLE
            else it.visibility = View.GONE
        }

    }

    override fun onClick(v: View) {
        dialog?.run {
            when (v.id) {
                R.id.tv_ok -> {
                    onOk.invoke()
                    dismiss()
                }
                R.id.tv_cancel -> {
                    onCancel.invoke()
                    dismiss()
                }
            }
        }

    }

    fun showBtnCancel(showCancel: Boolean): PrimaryDialog {
        this.showCancel = showCancel
        return this
    }
    fun showTitle(showTitle: Boolean): PrimaryDialog {
        this.showTitle = showTitle
        return this
    }
    fun setTitle(title: String): PrimaryDialog {
        this.title = title
        return this
    }
    fun setDescription(description: String): PrimaryDialog {
        this.description = description
        return this
    }

    fun show(fragmentManager: FragmentManager?) {
        fragmentManager ?: return
        show(fragmentManager, tag)
    }

}